pkg/ansible/proxy/proxy.go

```diff
@@ -411,18 +424,54 @@ func addWatchToController(owner metav1.OwnerReference, cMap *ControllerMap, reso
 		Version: gv.Version,
 		Kind:    owner.Kind,
 	}
-	c, watch, ok := cMap.Get(gvk)
+	contents, ok := cMap.GetController(gvk)
 	if !ok {
 		return errors.New("failed to find controller in map")
 	}
 	u := &unstructured.Unstructured{}
 	u.SetGroupVersionKind(gvk)
 	// Add a watch to controller
-	if watch {
-		log.Info("Watching child resource", "kind", resource.GroupVersionKind(), "enqueue_kind", u.GroupVersionKind())
-		err = c.Watch(&source.Kind{Type: resource}, &handler.EnqueueRequestForOwner{OwnerType: u})
-		if err != nil {
-			return err
+	if contents.WatchDependentResources {
+		// Use EnqueueRequestForOwner unless user has configured watching cluster scoped resources
+		if useOwnerReference && !contents.WatchClusterScopedResources {
+			log.Info("Watching child resource", "kind", resource.GroupVersionKind(), "enqueue_kind", u.GroupVersionKind())
+			err = contents.Controller.Watch(&source.Kind{Type: resource}, &handler.EnqueueRequestForOwner{OwnerType: u})
```

Comment:
Need to determine if we should watch via before attempting to watch and we need to store that we are watching this resource for the controller.